### ISSUE TITLE
Lua Error: C_Item.IsEquippableItem()

### ISSUE THREAD

**Issue by brittyazel** [Peterodox/YUI-Dialogue#36]
I had this error pop up today when I spoke to Therazal to pick up the weekly dragonflight quest. Note, this issue doesn't 'always' happen, and this seems to be a weird fluke. Other times I have spoken with this NPC have not resulted in the error, so I don't know what triggered it.

I am running WoW 11.0.2, with DialogueUI 0.4.0

```
1x bad argument #1 to '?' (Usage: local result = C_Item.IsEquippableItem(itemInfo))
[string "=[C]"]: in function `IsEquippableItem'
[string "@DialogueUI/API.lua"]:1930: in function <DialogueUI/API.lua:1926>
[string "=(tail call)"]: ?
[string "@DialogueUI/API.lua"]:1953: in function `IsRewardItemUpgrade'
[string "@DialogueUI/Code/SharedUITemplates.lua"]:1476: in function `SetItem'
[string "@DialogueUI/Code/SharedUITemplates.lua"]:1488: in function `?'
[string "@DialogueUI/Code/UI.lua"]:2641: in function `FormatRewards'
[string "@DialogueUI/Code/UI.lua"]:1393: in function `?'
[string "@DialogueUI/Code/UI.lua"]:2015: in function `ShowUI'
[string "@DialogueUI/Code/Core.lua"]:113: in function <DialogueUI/Code/Core.lua:80>

Locals:
(*temporary) = "bad argument #1 to '?' (Usage: local result = C_Item.IsEquippableItem(itemInfo))"

```

**Comment by fubaWoW** [comment]
Same here with the Quest [Training Our Forces](https://www.wowhead.com/quest=60388) from [Lord Herne](https://www.wowhead.com/npc=158544) Ardenweald, a Shadowlands Covenant Daily Quest.

**Comment by MechaZao** [comment]
I had it pop up doing the Dryad quest chain in the Emerald Dream. I also noticed that the text was corrupting and overlapping when it happened, and I would have to ESC out and click the questgiver again.

**Comment by Peterodox** [comment]
It should? be fixed now, by v0.4.1. I'll leave this ticket open in case it didn't, and for people who haven't installed the latest version.

I'm guessing the reward item's link may not always be available immediately, so when the addon starts to check if the item is an upgrade, it bugged out 😢
